### PR TITLE
Add resource function type definition

### PR DESCRIPTION
**Pull request by morganmspencer**
Adds the resource function type definition to the Server class

### CODE REVIEW

**Comment by cah-brian-gantzler** [comment]
Hoping @IanVS can look at this. I am not using typescript yet

File: types/index.d.ts [hunk 1]
```diff
@@ -322,6 +322,15 @@ declare module "miragejs/server" {
     /** A number of ms to artificially delay responses to this route. */
     timing?: number;
   }
+  
+  export interface ResourceOptions {
+    /** Whitelist of shorthand options */
+    only?: string[];
+    /** Exclude list of shorthand options */
+    except?: string[];
```
Comment (IanVS) [inline types/index.d.ts:330]: We can be a bit more helpful here, and use a union of the allowed values.

The values allowed are `'index' | 'show' | 'create' | 'update' | 'delete'`

File: types/index.d.ts [hunk 2]
```diff
@@ -452,6 +461,9 @@ declare module "miragejs/server" {
       handler?: RouteHandler<Registry, Response>,
       options?: HandlerOptions
     ): void;
+    
+    /** Define multiple shorthands for a given resource */
+    resource(model: string, options?: ResourceOptions): void;
```
Comment (IanVS) [inline types/index.d.ts:466]: Let's restrict the modelName to valid values, like what's done in `createList` above.  

```suggestion
    resource<K extends keyof Registry>(modelName: K, options?: ResourceOptions): void;
```

**Review by IanVS — CHANGES_REQUESTED** [review]
Thanks for the PR!  I made a few suggestions to help narrow down the typing a bit, and would you mind adding a test in https://github.com/miragejs/miragejs/blob/master/types/tests/server-test.ts as well?

**Review by IanVS — APPROVED** [review]

**Pull request merged**